### DISCUSSION TITLE
A trivial math question

### DISCUSSION THREAD

**Discussion by kironia** [kironia/99balloons#1] [Q&A] [answered]
I have a trivial math question, although things might get tricky in Java. Given two variables, one of type Integer, and the second of type Short. when I add them and cast the result to a String object, then what happens? Does it throw a class cast exception in this particular case?

**Comment by Stagyrite** [comment, marked as answer]
Of course, it won't throw a class cast exception. Compilers or a Java course can help to understand that. I recommend testing it out in The Java Shell tool (JShell). It's a tool available in modern Java.
As one can see, it wouldn't even compile because no integer can be cast to a String. Those two variables sum to an integer, and therefore can't be cast to a String.

```
|  Welcome to JShell -- Version 17.0.10
|  For an introduction type: /help intro

jshell> Short one = 1;
one ==> 1

jshell> Integer two = 2;
two ==> 2

jshell> String three = (String)(one + two);
|  Error:
|  incompatible types: int cannot be converted to java.lang.String
|  String three = (String)(one + two);
|                     ^---------^
```

The same can be shown with a Java class.
```java
public class TypeTest {

    public static void main(String []args) {
        Short one = 1;
        Integer two = 2;
        String three = (String)(one + two);
    }

}
```

Let's try to compile it, and we'll see the compile-time error.
```
javac TypeTest.java
TypeTest.java:6: error: incompatible types: int cannot be converted to String
        String three = (String)(one + two);
                           ^
1 error
```

The expression can be converted from an integer to a string by wrapping using 'String.valueOf()'. Another way is to concatenate the result with an empty string. Here's how it looks in JShell.

```
|  Welcome to JShell -- Version 17.0.10
|  For an introduction type: /help intro

jshell> Short one = 1;
one ==> 1

jshell> Integer two = 2;
two ==> 2

jshell> String three = String.valueOf(1 + 2);
three ==> "3"

jshell> three = one + two + "";
three ==> "3"
```
